parser-typechecker/src/Unison/Codebase/Type.hs

```diff
@@ -59,6 +62,8 @@ data Codebase m v a = Codebase
     -- Note that it is possible to call 'putTypeDeclaration', then 'getTypeDeclaration', and receive @Nothing@, per the
     -- semantics of 'putTypeDeclaration'.
     getTypeDeclaration :: Reference.Id -> m (Maybe (Decl v a)),
+    -- | Get the type of a given decl.
+    getDeclType :: V2.Reference -> m CT.ConstructorType,
```

Comment:
This was added because `Convert.referent2to1` requires it, and this PR is the first time we use V2 branches (and therefore v2 referents) outside of the codebase package.

Reply:
Adding `U.Codebase.` modules in here feels a little weird, but maybe it's fine?

Reply:
Are you able to expand on why it feels "weird"? I'm not sure how to address that unless I know why it feels off. Or, do you have suggestions for an alternative way of doing things?

If we plan to use V2 types I don't see any reason we'd avoid exposing them from the codebase, esp. since the codebase is largely implemented over the V2 types at this point.